internal/mesh/internal/controllers/sidecarproxy/controller_test.go

```diff
@@ -861,104 +861,112 @@ func TestMeshController(t *testing.T) {
 	suite.Run(t, new(controllerTestSuite))
 }
 
-func requireExplicitDestinationsFound(t *testing.T, name string, tmplResource *pbresource.Resource) {
-	requireExplicitDestinations(t, name, tmplResource, true)
+func (suite *controllerTestSuite) requireExplicitDestinationsFound(t *testing.T, name string, tmplResource *pbresource.Resource) {
+	suite.requireExplicitDestinations(t, name, tmplResource, true)
 }
 
-func requireExplicitDestinationsNotFound(t *testing.T, name string, tmplResource *pbresource.Resource) {
-	requireExplicitDestinations(t, name, tmplResource, false)
+func (suite *controllerTestSuite) requireExplicitDestinationsNotFound(t *testing.T, name string, tmplResource *pbresource.Resource) {
+	suite.requireExplicitDestinations(t, name, tmplResource, false)
 }
 
-func requireExplicitDestinations(t *testing.T, name string, tmplResource *pbresource.Resource, found bool) {
+func (suite *controllerTestSuite) requireExplicitDestinations(t *testing.T, name string, tmplResource *pbresource.Resource, found bool) {
 	t.Helper()
 
-	var tmpl pbmesh.ProxyStateTemplate
-	err := tmplResource.Data.UnmarshalTo(&tmpl)
-	require.NoError(t, err)
-
-	// Check outbound listener.
```

Comment:
Here and below: No changes to the test logic other than removing redundant unmarshals and nesting the remaining code under the new `waitForProxyStateTemplateState` helper func